package io.opencensus.metrics.export;

/* JADX INFO: loaded from: classes2.dex */
public abstract class ExportComponent {
    public abstract MetricProducerManager getMetricProducerManager();

    public static ExportComponent newNoopExportComponent() {
        return new NoopExportComponent();
    }

    private static final class NoopExportComponent extends ExportComponent {
        private static final MetricProducerManager METRIC_PRODUCER_MANAGER = MetricProducerManager.newNoopMetricProducerManager();

        private NoopExportComponent() {
        }

        @Override // io.opencensus.metrics.export.ExportComponent
        public MetricProducerManager getMetricProducerManager() {
            return METRIC_PRODUCER_MANAGER;
        }
    }
}
